src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -284,6 +298,24 @@ public void Dispose()
             Dispose(true);
         }
 
+        internal bool IsEnoughMemory(long hostProcessSize, IEnumerable<long> workerChannelSizes, long memoryLimit)
+        {
+            if (memoryLimit <= 0)
+            {
+                return true;
+            }
+
+            // Checking memory before adding a new worker
+            long maxWorkerSize = workerChannelSizes.Max();
+            long currentMemoryConsumption = workerChannelSizes.Sum() + hostProcessSize;
+            if (currentMemoryConsumption + maxWorkerSize > memoryLimit * 0.8)
```

Comment:
A comment explaining the choice of `0.8` would be nice. Should it be configurable at all as well?

Reply:
I added the comment to clarify, 80% of memory consumption threshold looks reasonable as we have the same limit for memory throttling in web jobs sdk:
https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Scale/ConcurrencyOptions.cs#L27

Also we do not want to introduce a new configurable setting which usefulness is doubtful.